### ISSUE TITLE
[Bug]: error: timeout waiting for connserver to register

### ISSUE THREAD

**Issue by juergenbarth** [wavetermdev/waveterm#987]
### Current Behavior

When trying to connect a tab to a Synology NAS via ssh, I get the following error:

Disconnected from "user@hostname.of.nas"
error: conncontroller user@hostname.of.nas start wsh connserver error: timeout waiting for connserver to register

No connection is established.

### Expected Behavior

The program should connect to the server without any problems, as it does with my Ubuntu based Linux server.

### Steps To Reproduce

1. Create a new tab
2. Create a terminal block
3. Click on the Laptop-Icon (Connect to Local Machine)
4. In the "Connect to" field enter either the hostname or the IP-address of the host to connect to, e.g. user@hostname.of.nas
5. Watch the error message come up

### Wave Version

Client Version 0.8.8 (202410012200)

### OS

macOS 14.7

### Architecture

arm64

### Questionnaire

- [X] I'm interested in fixing this myself but don't know where to start
- [ ] I would like to fix and I have a solution
- [ ] I don't have time to fix this right now, but maybe later

**Comment by esimkowitz** [comment]
@juergenbarth, what kind of authentication does your Synology server use for SSH?

**Comment by juergenbarth** [comment]
It uses ssh-key.
This used to work under version 0.7.

**Comment by esimkowitz** [comment]
Do you have a password or a passphrase on your key? Is it stored in your SSH config or in an SSH Agent?

Also do you see anything weird in ~/.waveterm/waveapp.log when you try to connect?

**Comment by esimkowitz** [comment]
@oneirocosm any ideas?

**Comment by esimkowitz** [comment]
@juergenbarth do you know if your NAS allows for mounting domain sockets over SSH? We use this mechanism to establish a connection between our app and your remote server.

Also, what OS does Synology use? Is it Linux?

**Comment by juergenbarth** [comment]
@esimkowitz All I can tell you is that this used to work with WaveTerm 0.7 and that I can log on this way directly from macOS Terminal via the ssh user@hostname.of.nas command.

The Synology OS is a Linux derivative.

**Comment by esimkowitz** [comment]
I wonder if the OS has some safeguard that is preventing us from running our WSH helper binary on the remote. At the moment, all connections established via our connections UI require our WSH helper to be loaded onto the remote before we complete the connection

**Comment by esimkowitz** [comment]
Though we had a similar mechanism in the old app so it'd be weird if that were the regression...

**Comment by juergenbarth** [comment]
Could it be related to the shell on the Synology? We only have sh not bash or zsh...

**Comment by esimkowitz** [comment]
@juergenbarth do you see the wsh binary in ~/.waveterm/bin on your Synology NAS?

**Comment by juergenbarth** [comment]
@esimkowitz Yes it's there, the permissions are 777. I can execute it, it brings the usage info. It even gets updated to the latest release

**Comment by oneirocosm** [comment]
That's really interesting. What about the `/tmp` directory? Does it exist? And if it does, what permissions does it have?

**Comment by juergenbarth** [comment]
@oneirocosm /tmp does exist and has permissions drwxrwxrwt.

**Comment by oneirocosm** [comment]
I'm still not completely sure why the domain socket isn't connecting, but we found a different bug which may end up resolving this one. I'll keep you posted as we work on the fix.

**Comment by oneirocosm** [comment]
I just realized i was ambiguous about something. The /tmp dir that exists with drwxrwxrwt, was that on your host or on the NAS? I meant to have you check on the NAS. Also, if it does exist on the NAS, is it possible for you to create a file there while logged in as the user you are trying to connect to?

**Comment by juergenbarth** [comment]
@oneirocosm Yes it is on the NAS:

```
me@nas:~$ touch /tmp/test
me@nas:~$ ll /tmp/test
-rw------- 1 me users 0 Oct 19 09:04 /tmp/test
me@nas:~$ rm /tmp/test
```

**Comment by oneirocosm** [comment]
Hello again! We recently encountered something similar to this on an Alpine Linux machine. The cause there was that in the machine's sshd config file (/etc/ssh/sshd_config), there was a line `AllowTcpForwarding no`. This prevented the unix domain socket from being sent to the host.

If that is the case for you, one way to fix it would be to set
`AllowTcpForwarding yes`

This will allow `wsh` to properly communicate across machines.

If you don't want to open tcp forwarding, or if you have a different issue, the next release should have a fix so you can connect to the machine without `wsh`, although in that case, you won't be able to use wsh commands with that connection.

We'd also like to make a longer term solution that keeps wsh in cases like this (we could do this by providing an alternative to unix domain sockets), but I don't know when that will be ready.

**Comment by joaberg** [comment]
I had the same problem on a RHEL server. Connection was just hanging, and never timing out.
Setting "AllowTcpForwarding yes" on the server and restarting sshd, then restarted waveterm to clear the hanging connection.
Now it works... thanks for the tip.

You should add this to a Q&A/Troubleshooting section in the docs.
